### PR TITLE
[grpc][Python][doc] Added CONTRIBUTING.md to guide new contributors through the process of contributing to gRPC Python

### PR DESCRIPTION
**Pull request by sourabhsinghs**
This PR introduces a new CONTRIBUTING.md file to the gRPC Python repository. The goal of this document is to provide clear and comprehensive guidelines for new contributors who are interested in contributing to the project.

- **Code of Conduct**: Establishes expectations for respectful and inclusive interactions within the community.
- **Guidelines**: Offers advice and best practices for contributing effectively, including how to find issues, get started with the codebase, and follow the project's standards.
- **Ways to Contribute**: Highlights the diverse ways in which people can contribute beyond just code, including documentation, community support, and testing.
- **Steps to Contributing**: Provides a step-by-step guide on how to set up a development environment, make changes, run tests, and submit a pull request.
- **Code Style**: Outlines the coding conventions and tools used in the project to ensure consistency.

### CODE REVIEW

File: src/python/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Contributing to gRPC Python
+We're thrilled you're interested in contributing to gRPC Python!  Our vibrant community is the heart of this project. Your expertise and ideas are invaluable, so join us in shaping the future of gRPC Python.
+
+## Legal Requirements
+**Sign the CLA:** Before your PR can be merged, you'll need to sign the [CNCF Contributor License Agreement (CLA)](https://identity.linuxfoundation.org/projects/cncf).
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:5]: We should change to `Before your PR can be reviewed`.

File: src/python/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+* **Code:** Fix bugs, add new features, or improve existing code.
+* **Documentation:** Improve tutorials, guides, or API reference documentation.
+* **Community:** Answer questions on forums, help triage issues, or write blog posts.
+* **Testing:** Help ensure the quality of gRPC Python by writing tests or reporting bugs.
+
+
+## Writing Your First Patch for gRPC Python
+
+Ready to contribute to gRPC Python? We've got you covered.  This guide will walk you through the entire process of making your first contribution, from identifying an issue to submitting your changes for review. Don't worry if you're new to open source – our documentation and helpful community will ensure a smooth experience.
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:24]: Should we rephrase this part? Using `This guide` here doesn't seems correct.

File: src/python/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+### Prerequisites
+
+* **Git:** You should have Git installed on your system. If not, download and install it from the [official Git website](https://git-scm.com/).
+* **GitHub Account:** You'll need a GitHub account to fork the repository and submit pull requests. 
+* **Python:** You should have a good understanding of Python programming. If you're new to Python, there are many resources available online to get you started. You can find the official documentation on the [Python website](https://www.python.org/doc/).
+* **gRPC Concepts:** Familiarize yourself with the basics of gRPC, including concepts like protocol buffers, services, clients, and servers. Refer to the [gRPC documentation](https://grpc.io/docs/) for an overview.
+* **Bazel:** Bazel is the build system used for gRPC. To install it, follow the instructions for your operating system on the [Bazel website](https://bazel.build/install).
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:32]: one of the build systems?

File: src/python/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+### Steps to Contributing to gRPC Python
+
+1. **Find an Issue:**
+   * **Browse Open Issues:** Look for issues labeled "[help wanted](https://github.com/grpc/grpc/issues?q=is%3Aopen+label%3A%22disposition%2Fhelp+wanted%22)" on the [gRPC Python issue tracker](https://github.com/grpc/grpc/issues?q=is%3Aissue+is%3Aopen+label%3Alang%2Fpython+sort%3Aupdated-desc).
+   * **Ask for Help:** If you're unsure where to start or need clarification on an issue, feel free to ask questions on the [gRPC forum](https://groups.google.com/g/grpc-io).
+2. **Get a Copy of the gRPC Python Development Version:**
+   * **Fork:**  Click the "Fork" button on the top right of the [gRPC repository page](https://github.com/grpc/grpc) to create a copy of the repository under your GitHub account.
+   * **Clone your fork:**
+   ```git clone https://github.com/<your-username>/grpc.git```
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:42]: I think we also need initialize and update submodules: `git submodule update --init --recursive`

File: src/python/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+3. **Setting up Your Local System for Development and Testing:**
+   * Create a new virtual environment by running:```python -m venv ~/.virtualenvs/grpc-python```
+   * Activate the environment: ```source ~/.virtualenvs/grpc-python/bin/activate```
+   * Install Dependencies: ```pip install -r requirements.txt```
+4. **Run Tests:**
+   * Before making any changes, run the existing test suite to ensure your environment is set up correctly:
+   * **Using Bazel (Recommended):**
+     * To run a single unit test:
+       ```bash
+       bazel test --cache_test_results=no --test_output=streamed --runs_per_test=1 --test_timeout=50 "//src/python/grpcio_tests/tests/unit:_abort_test"
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:54]: Consider remove `--test_output=streamed --runs_per_test=1 --test_timeout=50`? Seems to be too many flags for first time users.

File: src/python/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+     * To run a single unit test:
+       ```bash
+       bazel test --cache_test_results=no --test_output=streamed --runs_per_test=1 --test_timeout=50 "//src/python/grpcio_tests/tests/unit:_abort_test" 
+       ```
+     * To execute all unit tests for Python:
+       ```bash
+       bazel test --cache_test_results=no --test_output=streamed --runs_per_test=1 --test_timeout=50 "//src/python/..." 
+       ```
+   * **Using Provided Scripts (Alternative):**
+     * **Install Python Modules:**
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:61]: We should keep format the same, either use **Bold** case above or remove bold case here.

File: src/python/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+7. **Code Review:**
+    * **Wait for Feedback:** Maintainers will review your PR and provide feedback.
+    * **Respond to Comments:**  Address any questions or concerns raised during the review.
+    * **Make Revisions:** Update your code as needed based on the feedback.
+
+## Code Style
+
+* **Pythonic Code:** Follow the [PEP 8 style guide](https://www.python.org/dev/peps/pep-0008/) for Python code.
+* **Type Hints:** Use type hints to improve code readability and maintainability.
+* **Linting:** We use [Flake8](http://flake8.pycqa.org/en/latest/) for code linting. Make sure your code passes Flake8 checks.
```
Comment (XuanWang-Amos) [inline src/python/CONTRIBUTING.md:88]: Hmm are we using Flake8?

File: src/python/CONTRIBUTING.md (hunk 6 above)
Comment (XuanWang-Amos) [reply src/python/CONTRIBUTING.md:66]: About this, I mean remove `**` used in `Install Python Modules:` and `Run Tooling Tests:`

**Review by XuanWang-Amos — APPROVED** [review]
Only one comment left, thanks for working on this!

File: src/python/CONTRIBUTING.md (hunk 6 above)
Comment (sourabhsinghs) [reply src/python/CONTRIBUTING.md]: removed ** used in Install Python Modules: and Run Tooling Tests:

**Pull request closed without merging**